### PR TITLE
[ODS-6165] Update Actions Workflows for main-6x

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

**Comment by stephenfuqua** [comment]
The build failures we see here  are expected, since we haven't finished updating everything to .NET 8. If the code in the pull request looks fine and is approved through code review, then I will apply an administrative override to allow the squash & merge.

File: .github/workflows/Dependencies Security Scan.yml [hunk 1]
```diff
@@ -7,17 +7,18 @@ name: Dependency Review Pull request
...
   HEAD_REF:  ${{ GITHUB.HEAD_REF }}
   REF_NAME:  ${{ GITHUB.REF_NAME }}
-  REPOSITORY_DISPATCH_BRANCH: ${{ github.event.client_payload.branch }}  
+  REPOSITORY_DISPATCH_BRANCH: ${{ github.event.client_payload.branch }}
 
 jobs:
   scan-actions:
     name: Scan Actions
-    uses: ed-fi-alliance-oss/ed-fi-actions/.github/workflows/repository-scanner.yml@latest
+    uses: ed-fi-alliance-oss/ed-fi-actions/.github/workflows/repository-scanner.yml@main
```
Comment (stephenfuqua) [inline .github/workflows/Dependencies Security Scan.yml:21]: We changed from using `latest` to `main` sometime last year, as it is easier to manage that way.

File: .github/workflows/Dependencies Security Scan.yml [hunk 2]
```diff
@@ -31,17 +32,8 @@ jobs:
...
       - name: Checkout Ed-Fi-ODS
-        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2
-        with:
-            repository: Ed-Fi-Alliance-OSS/Ed-Fi-ODS
-            path: Ed-Fi-ODS/
-      - name: Is pull request branch exists in Ed-Fi-ODS
-        working-directory: ./Ed-Fi-ODS/
-        shell: pwsh
-        run: |
-             .\build.githubactions.ps1 CheckoutBranch -RelativeRepoPath "."
```
Comment (stephenfuqua) [inline .github/workflows/Dependencies Security Scan.yml:45]: Dependency review also runs in the Ed-Fi-ODS-Implementation repository, in its own workflow. Therefore we don't need _this_ job to look at both repositories: it is safe to only look at the current repository.

File: Utilities/DataLoading/EdFi.LoadTools.Test/EdFi.LoadTools.Test.csproj
Comment (stephenfuqua) [inline Utilities/DataLoading/EdFi.LoadTools.Test/EdFi.LoadTools.Test.csproj:1]: All tests are passing on my system. It is necessary to start ODS/API from `v6.1-patch2` branch to run these tests. You also need to copy the correct credentials from the sandbox admin appSettings.developer.json into the appsettings.json file in this repository (but don't accidentally commit that change).

image

**Review by JBrenesSimpat — APPROVED** [review]

File: Utilities/DataLoading/EdFi.LoadTools/EdFi.LoadTools.csproj [hunk 3]
```diff
@@ -22,14 +22,15 @@
     <None Include="../../../LICENSE.txt" Pack="true" PackagePath="LICENSE.txt" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="Autofac" Version="6.3.0" />
-    <PackageReference Include="EdFi.Suite3.Common" Version="6.1.16" />
+    <PackageReference Include="Autofac" Version="8.0.0" />
+    <PackageReference Include="EdFi.Suite3.Common" Version="[6.2.2,7.0.0)" />
```
Comment (semalaiappan) [inline Utilities/DataLoading/EdFi.LoadTools/EdFi.LoadTools.csproj:26]: I didn't understand this [6.2.2,7.0.0)  why we have 2 versions and also different closing bracket ?

Comment (stephenfuqua) [reply Utilities/DataLoading/EdFi.LoadTools/EdFi.LoadTools.csproj:26]: Great question! This tells the package manager that the minimum version is 6.2.2, and that it should not try to update to any version starting with 7.0.0. This helps avoid accidental updates or downgrades to an inappropriate version.

Comment (stephenfuqua) [reply Utilities/DataLoading/EdFi.LoadTools/EdFi.LoadTools.csproj:26]: For more information on setting version ranges, see https://learn.microsoft.com/en-us/nuget/concepts/package-versioning?tabs=semver20sort#version-ranges

**Pull request merged**